odoo/api.py

```diff
@@ -631,25 +631,41 @@ def __call__(self, cr=None, user=None, context=None, su=None):
         su = (user is None and self.su) if su is None else su
         return Environment(cr, uid, context, su, self.uid_origin)
 
-    def ref(self, xml_id, raise_if_not_found=True):
-        """ Return the record corresponding to the given ``xml_id``.
+    def ref(self, *xml_ids: str, raise_if_not_found: bool = True) -> BaseModel | None:
+        """ Return the records corresponding to the given ``xml_ids``.
 
-        :param str xml_id: record xml_id, under the format ``<module.id>``
+        :param str xml_ids: record xml_ids, under the format ``<module>.<record_id>``
         :param bool raise_if_not_found: whether the method should raise if record is not found
-        :returns: Found record or None
-        :raise ValueError: if record wasn't found and ``raise_if_not_found`` is True
+        :returns: The recordset found with given xml_ids
+        :raise ValueError: if some records weren't found and ``raise_if_not_found`` is True,
+                           or if the user mixed records belonging to different models.
         """
-        res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
-            xml_id, raise_if_not_found=raise_if_not_found
-        )
 
-        if res_model and res_id:
-            record = self[res_model].browse(res_id)
-            if record.exists():
-                return record
-            if raise_if_not_found:
-                raise ValueError('No record found for unique ID %s. It may have been deleted.' % (xml_id))
-        return None
+        id_xmlid = {}
+        first_model_name, first_xml_id = None, None
+        for xml_id in xml_ids:
+            res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
+                xml_id, raise_if_not_found=raise_if_not_found
+            )
+            if res_model and res_id:
+                if not first_model_name:
+                    first_model_name = res_model
+                    first_xml_id = xml_id
+                elif first_model_name != res_model:
+                    raise ValueError(
+                        f"'{first_xml_id}' and '{xml_id}' xmlids belong to different models."
+                    )
+                id_xmlid[res_id] = xml_id
+        if not id_xmlid:
+            return None
+
+        records = self[first_model_name].browse(id_xmlid)
+        if raise_if_not_found and (missing := records - records.exists()):
```

Comment:
Sorry, this logic is wrong 😢 
Prevous: always return existing reocrds but only raise error when raise_if_not_found
Now: when raise_if_not_found return existing records, otherwise return all existing and non-existing records